### ISSUE TITLE
Surface peek memory usage since restart

### ISSUE THREAD

**Issue by umanwizard** [MaterializeInc/materialize#17364]
### Feature request

This is probably a useful thing to show to customers.

**Comment by tokenrove** [comment]
I like `getrusage`'s `ru_maxrss` because it's truly the highest your process's RSS has ever been (as observed by the kernel), but from discussion, it may be difficult to plumb this through.  cgroups v2 exposes `memory.peak` which is not obviously the same thing but may be close enough, and presumably kubernetes should allow exposing that on a per-pod basis.

**Comment by umanwizard** [comment]
Can you summarize that discussion?

**Comment by benesch** [comment]
To the extent possible, I'd prefer to keep monitoring of memory and CPU fully external to the monitored possible. Especially when OOM kills are in the cards, its nice for the process under observation not to be involved in that observation, as otherwise you're basically guaranteed to miss the metrics in the moments before process exit.

We should be well set up for this now that we have the Prometheus adapter installed. I _think_ it should be straightforward to configure the adapter to scrape the `container_memory_max_usage_bytes` metric from cAdvisor (see https://github.com/google/cadvisor/blob/master/docs/storage/prometheus.md), though I have to confess I don't have the full stack of metrics tools paged in.
